rapidy/_annotation_extractor.py

```diff
@@ -1,69 +1,256 @@
+class ParameterCannotHaveDefaultError(Exception):
+    _base_err_msg = 'Handler attribute with Type `{class_name}` cannot have a default value.'
+
+    def __init__(self, *args: Any, class_name: str, handler: Any, param_name: str) -> None:
+        super().__init__(
+            f'{self._base_err_msg.format(class_name=class_name)}\n{_create_handler_attr_info_msg(handler, param_name)}',
+            *args,
+        )
+
+
+class ParameterCannotHaveDefaultFactoryError(Exception):
+    _base_err_msg = 'Handler attribute with Type `{class_name}` cannot have a default_factory.'
+
+    def __init__(self, *args: Any, class_name: str, handler: Any, param_name: str) -> None:
+        super().__init__(
+            f'{self._base_err_msg.format(class_name=class_name)}\n{_create_handler_attr_info_msg(handler, param_name)}',
+            *args,
+        )
+
+
+class SpecifyBothDefaultAndDefaultFactoryError(TypeError):
+    _base_err_msg = 'Cannot specify both default and default_factory in `{class_name}`.'
+
+    def __init__(self, *args: Any, class_name: str, handler: Any, param_name: str) -> None:
+        super().__init__(
+            f'{self._base_err_msg.format(class_name=class_name)}\n{_create_handler_attr_info_msg(handler, param_name)}',
+            *args,
+        )
+
+
+class IncorrectDefineDefaultValueError(Exception):
+    _base_err_msg = (
+        'Default value cannot be set in `{class_name}`. '
+        'You cannot specify a default value using Param(<default_value>, ...) and `=` at the same time.'
+    )
+
+    def __init__(self, *args: Any, class_name: str, handler: Any, param_name: str) -> None:
+        super().__init__(
+            f'{self._base_err_msg.format(class_name=class_name)}\n{_create_handler_attr_info_msg(handler, param_name)}',
+            *args,
+        )
+
+
+class UnsupportedSchemaDataTypeError(TypeError):
+    def __init__(self, *args: Any, err_msg: str, handler: Any, param_name: str) -> None:
+        super().__init__(
+            f'{err_msg}\n{_create_handler_attr_info_msg(handler, param_name)}',
+            *args,
+        )
+
+
+def _get_annotation_data_by_annotated_flow(
+        attr_type: Any,
+        param_field_info: ParamFieldInfo,
+        handler: Handler,
+        param: inspect.Parameter,
+) -> Tuple[Any, ParamFieldInfo, Any]:
+    default = _get_annotated_definition_attr_default(handler=handler, field_info=param_field_info, param=param)
+    return attr_type, param_field_info, default
+
+
+def _get_annotation_data_by_default_value_flow(
+        handler: Handler,
         param: inspect.Parameter,
-) -> Tuple[Any, ParamFieldInfo]:
-    if get_origin(param.annotation) is not Annotated:
-        raise UnknownParameterError
+) -> Tuple[Any, ParamFieldInfo, Any]:
+    field_info = _prepare_field_info(param.default)
+    default = _get_default_definition_attr_default(handler=handler, field_info=field_info, param_name=param.name)
+    return param.annotation, field_info, default
```

Comment:
TODO: let's use some structure here, not a raw tuple to remove that code.
```python
if len(annotated_args) != 2:
    raise NotParameterError
 ```

Reply:
This structure is a crutch because I don't know how to more accurately handle the scenarios related to annotations and metainfo in this block. Any suggestions?

```py
if len(annotated_args) != 2:
    raise NotParameterError
```

Reply:
can you look again? I added response schema